korra/ArmActor.py

```diff
@@ -0,0 +1,56 @@
+
+import pykka
+import time
+
+from pickit.Datatypes import *
+from pickit import ArmManager
+
+from korra.ArmUtils import ArmWrapper, joint_states_to_arm
+
+class ArmActor(pykka.ThreadingActor):
+    def __init__(self, state_publisher, environment, estimated_delay, flip):
+        super(ArmActor, self).__init__()
+        self.estimated_delay = estimated_delay
+        self.state_publisher = state_publisher
+        self.environment = environment
+        self.flip = flip
+
+        # Get initial state of arm to initialise Arm
+        now = time.time()
+        (q0, q0_vel) = self.get_arm_state(now)
+        self.arm = ArmWrapper(q0, flip)
+
+    def on_receive(self, msg):
+        if msg.get('cmd') == 'send_right_arm_traj':
+            self.send_new_traj()
+
+    def send_new_traj(self):
+        now = time.time()
+        state = self.get_arm_state(now) # state is pos & vel JointSpacePoint
```

Comment:
maybe consider replanning from 'now + self.estimated_delay' since in general you're moving while replanning